fenix/app/src/test/java/org/mozilla/fenix/gleanplumb/state/MessagingMiddlewareTest.kt

```diff
@@ -48,240 +38,226 @@ class MessagingMiddlewareTest {
     @get:Rule
     val coroutinesTestRule = MainCoroutineRule()
     private val coroutineScope = coroutinesTestRule.scope
-    private lateinit var appStore: AppStore
-    private lateinit var middleware: MessagingMiddleware
     private lateinit var messagingStorage: NimbusMessagingStorage
-    private lateinit var messagingController: NimbusMessagingController
-    private lateinit var middlewareContext: MiddlewareContext<AppState, AppAction>
+    private lateinit var controller: NimbusMessagingController
 
     @get:Rule
     val gleanTestRule = GleanTestRule(testContext)
 
     @Before
     fun setUp() {
-        appStore = mockk(relaxed = true)
         messagingStorage = mockk(relaxed = true)
-        messagingController = spyk(NimbusMessagingController(messagingStorage) { 0L })
-        middlewareContext = mockk(relaxed = true)
-        every { middlewareContext.store } returns appStore
-
-        middleware = MessagingMiddleware(
-            messagingStorage,
-            messagingController,
-            coroutineScope,
-        )
+        controller = NimbusMessagingController(messagingStorage) { 0 }
     }
 
     @Test
-    fun `WHEN Restore THEN get messages from the storage and UpdateMessages`() = runTestOnMain {
-        val messages: List<Message> = emptyList()
+    fun `WHEN restored THEN get messages from the storage`() = runTestOnMain {
+        val store = AppStore(
+            AppState(
+                messaging = MessagingState(
+                    messages = emptyList(),
+                ),
+            ),
+            listOf(
+                MessagingMiddleware(messagingStorage, controller, coroutineScope),
+            ),
+        )
 
-        coEvery { messagingStorage.getMessages() } returns messages
+        val message = Message(
+            "message1",
+            mockk(relaxed = true),
+            "",
+            StyleData(),
+            emptyList(),
+            Message.Metadata("same-id", displayCount = 0),
+        )
```

Comment:
So here is the other bit of refactoring we could do if you think it's worthwhile.

We create virtually identical messages for each test with only one/two changes. A nice way to make these tests easier to parse or easier to write more in the future is with simple helper functions for our tests:

```kotlin
private fun createMessage(
    messageId: String = "control-id",
    data: MessageData = mockk(relaxed = true),
    action: String = "action",
    styleData: StyleData = StyleData(),
    triggers: List<String> = listOf("triggers"),
    metadata: Message.Metadata = createMetadata(),
) = Message(messageId, data, action, styleData, triggers, metadata)

private fun createMetadata(
    id: String = "same-id",
    displayCount: Int = 0,
    pressed: Boolean = false,
    dismissed: Boolean = false,
    lastTimeShown: Long = 0L,
    latestBootIdentifier: String? = null,
) = Message.Metadata(
    id,
    displayCount,
    pressed,
    dismissed,
    lastTimeShown,
    latestBootIdentifier,
)
```

This means that this line or the ones below turn into one/two liners like this:

```suggestion
        val message = createMessage()
```

Reply:
Yes makes sense